AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_component_validation.py

```diff
@@ -0,0 +1,210 @@
+     {
+        "test_description": (x_float_value, y_float_value, z_float_value, expect_pass),
+        "Zero Value Test": (0.0, 0.0, 0.0, True)
+     }
+    """
+    # Dictionary Keys
+    x, y, z, expect_pass = 0, 1, 2, 3
+
+    Report.info(f"Validating {component_name}'s {property_name} Vector3 property can be set or fails gracefully.")
+
+    for test_name in tests:
+        test = tests[test_name]
+
+        set_vector3_value(test[x], test[y], test[z])
+
+        set_value = get_vector3_value()
+        expected_value = Math.Vector3(test[x], test[y], test[z])
+
+        assert compare_vec3(expected_value, set_value) is test[expect_pass], \
+            f"Error: The {component_name}'s  {property_name} property failed to the \"{test_name}\" test. " \
+            f"{expected_value} was expected but {set_value} was retrieved. " \
+            f"Negative Scenario Test: {not test[expect_pass]}."
+
+def validate_property_switch_toggle(property_name: str, get_toggle_value: typing.Callable,
+                                    set_toggle_value: typing.Callable, component_name: str,
+                                    restore_default: bool=True) -> None:
+    """
+    Used to toggle a property switch and validate that it toggled.
+    param component_property_path: String of component property. (e.g. 'Settings|Visible')
+
+    :return: None
+    """
+    Report.info(f"Validating {component_name}'s {property_name} property toggle switch toggles.")
+
+    start_value = get_toggle_value()
+    set_toggle_value(not start_value)
+
+    end_value = get_toggle_value
+
+    assert (start_value != end_value), \
+        f"Error: {component_name}'s {property_name} property toggle switch did not toggle to " \
+        f"{end_value} when it started at {start_value}"
+
+    if restore_default:
+        set_toggle_value(start_value)
+
+def validate_asset_property(property_name: str, get_asset_value: typing.Callable, set_asset_value: typing.Callable,
+                            component_name: str, asset_path: str) -> None:
+    """
+    Function to validate the behavior of a property that gets and sets an Integer value.
+
+    :get_float_value: The Editor Entity Component Property's get method.
+    :set_float_value: The Editor Entity Component Property's set method.
+    :component_name: The name of the Editor Entity Component under test.
+    :property_name: The name of the Editor Entity Component Property under test.
+    :tests: A dictionary that stores an Int value and the intent of the values being passed. The following
+    format is expected
+     {
+        "test_description": (int_value, expect_pass),
+        "Zero value Test": (0, True)
+     }
+    """
+    Report.info(f"Validating {component_name}'s {property_name} Asset property can be set.")
+
+    set_asset_value(asset_path)
+
+    set_value = get_asset_value()
+    expected_asset = Asset.find_asset_by_path(asset_path)
+
+    assert expected_asset.id.is_equal(expected_asset.id, set_value), \
```

Comment:
I played with the `AssetId.is_equal` function for a while because I made the same assumption.

```
class AssetId:
    def is_valid(self, arg0: AssetId) -> None:
        pass

    def to_string(self, arg0: AssetId) -> None:
        pass

    def is_equal(self, arg0: AssetId, arg1: AssetId) -> None:
        pass
```

Switching to `math.Uuid.Equal` would require some extra code overhead to make it work, so I'm going to stick with the current methodology.